src/WebUI/src/services/activity-logs-service.ts

```diff
@@ -0,0 +1,49 @@
+import qs from 'qs';
+import { UserPublic } from '@/models/user';
+import { ActivityLogType, type ActivityLog } from '@/models/activity-logs';
+import { get } from '@/services/crpg-client';
+import { getUsersByIds } from '@/services/users-service';
+
+export interface ActivityLogsPayload {
+  from: Date;
+  to: Date;
+  userId?: number[];
+  type?: ActivityLogType[];
+}
+
+export const getActivityLogs = async (payload: ActivityLogsPayload) => {
+  const params = qs.stringify(payload, {
+    strictNullHandling: true,
+    arrayFormat: 'brackets',
+    skipNulls: true,
+  });
+
+  return (await get<ActivityLog[]>(`/activity-logs?${params}`)).map(al => ({
+    ...al,
+    createdAt: new Date(al.createdAt),
+  }));
+};
+
+const extractUsersFromLogs = (logs: ActivityLog[]) =>
+  logs.reduce((out, l) => {
+    if (l.type === ActivityLogType.TeamHit) {
```

Comment:
What do you think about looking for know metadata key such as 'targetUserId'. That would be more future proof